### PR TITLE
Vijay Anand - Hotfix for assign badge button and search field bug on entering space

### PR DESCRIPTION
**Pull request by vijay10pk**
# Description
 
Screenshot 2024-09-02 at 1 27 25 PM

**Issues**
Assign badge Button:
Screenshot 2024-09-02 at 1 26 51 PM

Search Field:
Screenshot 2024-09-02 at 1 27 31 PM

## Related PRS (if any):
To test this frontend PR you need to check out the development branch for the backend.
…

## Main changes explained:
- Refactored the code to remove any unnecessary space in the beginning and trailing of the search input.
- Refactored the code so that the "Assign Badge" button will be enabled only when a user is selected.
…

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. Clear site data/cache
4. log as admin/owner user
5. go to dashboard→ Other Links→ Badge Management → Badge Assignment(Tab)
6. verify if the user can see the search result even if you enter space at the beginning or trailing of the name when entered in search field.
7. verify if the "Assign Badge" button is enabled only when a user name is selected from the search result.

## Screenshots or videos of changes:
https://www.loom.com/share/d21ba9e1c97c419ab512b8b5b8f2468f?sid=a000250c-97c0-4c5c-8f4f-f3e24658c242

## Note:
This changes are related only to badge assignment component

### CODE REVIEW

**Comment by agryash** [comment]
This PR works as expected.

From what I noticed these are the two changes you made:
1. Trimming fullName to use trimmedName
2. Disabling the button if no selectedUserId is present

My suggestion would be to include error handling for the fullName field. If the fullName field gets passed something else other than a string, it might error out.

File: src/components/Badge/AssignBadge.jsx [hunk 1]
```diff
@@ -42,15 +42,16 @@ function AssignBadge(props) {
   }, []);
 
   useEffect(() => {
-    if (fullName) {
+    const trimmedName = fullName.trim(); // Trim any leading/trailing spaces
```
Comment (agryash) [inline src/components/Badge/AssignBadge.jsx:45]: Add error handling for fullName

**Review by manikittu810 — DISMISSED** [review]
Hello @vijay10pk 
Followed steps in the description. Everything is working as mentioned.
Screenshot 2024-09-03 at 10 26 52 PM

Thank you.

**Review by Vigneshwar-Muriki — DISMISSED** [review]
Tested this PR and its working as expected.


https://github.com/user-attachments/assets/afc1ce8f-8d22-4a1e-84d4-58408f176110

**Review by RishithaMamidala — CHANGES_REQUESTED** [review]
@vijay10pk 
The "Assign Badge" button is initially disabled, but when you search for a name and then clear the input, the button remains enabled even with no name present.
Chcek below video for walkthrough.

https://github.com/user-attachments/assets/f5cdd068-7901-47df-b713-61b5899f1154

**Review by beblicarl — CHANGES_REQUESTED** [review]
The search field works perfectly. However, the assign badge appears even when there is nothing in the search field

image

image

**Review by CillianRen — DISMISSED** [review]
image
Worked as expected. Good job.

**Comment by one-community** [comment]
Thank you all, merging!

**Review by nikhilpittala16 — APPROVED** [review]
I have tested the PR and everything works as expected.

https://github.com/user-attachments/assets/d59c2d4d-90c5-4e53-b3ad-cab56939683d

**Pull request merged**